vcr/request.py

```diff
@@ -32,9 +33,19 @@ def __init__(self, method, uri, body, headers):
             self.body = body.read()
         else:
             self.body = body
-        self.headers = {}
-        for key in headers:
-            self.add_header(key, headers[key])
+        self.headers = CaseInsensitiveDict()
```

Comment:
I honestly think the current behavior of `add_header` is pretty weird, but given that this the case, the setter SHOULD also do the same thing, right?

In general, I think its better to have logic for populating values like that live in exactly one place, and I think that the setter is the right place for that. The behavior should be abstracted into the dict object in any case.